Weather App/controller/ViewController.swift

```diff
@@ -97,6 +97,23 @@ extension ViewController: UITableViewDataSource {
 
         return cell
     }
+
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if weatherForCityViewModel.weatherForLocation != nil, indexPath == IndexPath(row: 0, section: 0) {
+            return
+        }
+
+        if editingStyle == .delete {
+            weatherForCityViewModel.removeWeather(index: indexPath.row)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+            let citiesNames = weatherForCityViewModel.weathers.map { $0.name }
+            do {
+                try storage.save(cities: citiesNames)
```

Comment:
I think, ViewModel should have `storage` property not ViewController and then you call only fetchWeathers or removeWeather directly from ViewModel. Saving new list of cities should be inside removeWeather functions in ViewModel